pkgs/test_core/lib/src/runner/vm/test_compiler.dart

```diff
@@ -0,0 +1,154 @@
+
+    import "package:test_core/src/bootstrap/vm.dart";
+
+    import "$testUri" as test;
+
+    void main(_, SendPort sendPort) {
+      internalBootstrapVmTest(() => test.main, sendPort);
+    }
+  ''';
+  }
+
+  // Handle a compilation request.
+  Future<void> _onCompilationRequest(_CompilationRequest request) async {
+    final isEmpty = _compilationQueue.isEmpty;
+    _compilationQueue.add(request);
+    if (!isEmpty) {
+      return;
+    }
+    while (_compilationQueue.isNotEmpty) {
+      final request = _compilationQueue.first;
+      var firstCompile = false;
+      CompileResult? compilerOutput;
+      final contents =
+          await _generateEntrypoint(request.mainUri, request.metadata);
+      final tempFile = File(p.join(_outputDillDirectory.path, 'test.dart'))
+        ..writeAsStringSync(contents);
+
+      if (_frontendServerClient == null) {
+        compilerOutput = await _createCompiler(tempFile.uri);
+        firstCompile = true;
+      } else {
+        compilerOutput =
+            await _frontendServerClient!.compile(<Uri>[tempFile.uri]);
+      }
+      // The client is guaranteed initialized at this point.
+      final client = _frontendServerClient!;
+      final outputPath = compilerOutput?.dillOutput;
+      if (outputPath == null) {
+        request.result.complete(CompilationResponse(
+            compilerOutput: compilerOutput?.compilerOutputLines.join('\n'),
+            errorCount: compilerOutput?.errorCount ?? 0));
+      } else {
+        final outputFile = File(outputPath);
+        final kernelReadyToRun = await outputFile.copy('${tempFile.path}.dill');
+        final testCache = File(_dillCachePath);
+        // Keep the cache file up-to-date and use the size of the kernel file
+        // as an approximation for how many packages are included. Larger files
+        // are prefered, since re-using more packages will reduce the number of
+        // files the frontend server needs to load and parse.
+        if (firstCompile ||
+            !testCache.existsSync() ||
+            (testCache.lengthSync() < outputFile.lengthSync())) {
+          if (!testCache.parent.existsSync()) {
+            testCache.parent.createSync(recursive: true);
+          }
+          await outputFile.copy(_dillCachePath);
+        }
+        request.result.complete(CompilationResponse(
+            compilerOutput: compilerOutput?.compilerOutputLines.join('\n'),
+            errorCount: compilerOutput?.errorCount ?? 0,
+            kernelOutputUri: kernelReadyToRun.absolute.uri));
+        client.accept();
+        client.reset();
+      }
+      _compilationQueue.removeAt(0);
+    }
+  }
+
+  Future<CompileResult?> _createCompiler(Uri testUri) async {
+    final platformDill = 'lib/_internal/vm_platform_strong.dill';
+    final sdkRoot =
+        Directory(p.relative(p.join(Platform.resolvedExecutable, '..', '..')))
+            .uri;
+    var client = _frontendServerClient = await FrontendServerClient.start(
```

Comment:
I believe I added an option to skip printing incremental dependencies - you should definitely turn that on to get another 20-30% speedup on large numbers of tests

Reply:
done 👍